packages/react-scripts/proxy/setupProxy.js

```diff
@@ -30,15 +30,30 @@ const setProxies = (app, customProxies = []) => {
   const target = process.env.BASE_URL
 
   const setProxy = proxyConfig => {
-    app.use(
-      proxy(proxyConfig.route, {
-        target,
-        changeOrigin: true,
-        logLevel: 'debug',
-        timeout: 5000,
-        ...proxyConfig.options,
+    const options = {
+      target,
+      changeOrigin: true,
+      logLevel: 'debug',
+      timeout: 5000,
+      ...proxyConfig.options,
+    }
+
+    if (proxyConfig.accept) {
+      app.use((req, res, next) => {
+        // proxy only if accept type starts with the same string
+        // (e.g., type 'application/' works for 'application/x-gedcomx-v1+json' and 'application/json')
+        if (req.headers.accept && req.headers.accept.indexOf(proxyConfig.accept) === 0) {
```

Comment:
The accept header can have multiple entries. For instance, I could ask for 'text/something' and then 'application/something else'. You are only catching the first instance here. Though this is certainly the norm, it is not what the specification says is possible.

Reply:
This is the way search-artifact and search-bifrost use it. I don't know of any other team that uses it, and we haven't needed anything more complex in 5 years, so I kept it simple.

Reply:
Its probably fine at this point. We've never needed anything more complex in the time we've used this logic.